README.md

```diff
@@ -1,8 +1,160 @@
+
+## Local
+After cloning this repo, get the required packages by running `npm install`.
+### Selenium
+To automate web browser actions, CodeceptJS requires a Selenium webserver. You have two options here: Docker or npm. Note that for both methods, you can visit `localhost:4444/wd/hub/sessions` to see current Selenium session/check that the server is running.
+#### Docker ([link](https://github.com/SeleniumHQ/docker-selenium))
+If you have docker, you can just run the preconfigured container
+```
+docker run -d -p 4444:4444 -v /dev/shm:/dev/shm selenium/standalone-chrome
+```
+To kill the server just kill the container.
+#### npm ([link](https://www.npmjs.com/package/selenium-standalone))
+If you'd rather not fool with docker, you can run the server yourself with an npm package. 
+If you already ran `npm install`, the package selenium-standalone should have been installed. You'll also need to install the webdriver, just run `npm run selenium-install`.
+
+You'll need the Java SDK, version 8, to use Selenium so make sure you have that installed as well (check output of `javac -version`, e.g. javac 1.8.0_171 means I have version 8.0_171)
+
+Once this is done, you can run `npm run selenium-start`, and the server will start running on port 4444 by default.
+
+You can kill the server with `npm run selenium-kill`
+
+### Data _(optional)_
+Some tests need data to submit, so this step is required only if you plan on running/writing tests that submit, delete, or query graph data (aka nodes). If your test doesn't need this stuff, just skip this step.
+Currently you have to clone the [occ/data-simulator](https://github.com/occ-data/data-simulator) repo, install R, install the required packages, and manually run the command for generating simulated data.
+
+The tests assume that data is simulated for project `test`, and that only 1 sample is simulated for each node type. Here's an example command of what that would look like if we were saving the data to `~/testData/`:
+```
+Rscript GenTestDataCmd.R https://my.dictionary.url/ test 1 ~/testData/
+```
+
+
+# Test Development
+## Running Tests
+Once everything is setup and your Selenium server is running, you should be able to successfully run the tests. Try running the following
+```
+npm test
+```
+If all the tests start running (and passing), everything should be good to go for you to start writing your own tests. 
+Note that you can also run a selection of tests with certain tags by running the following:
+```
+npm test -- --grep "@MyTag"
+```
+You can also run a specific suite by doing the following:
+```
+npm test ./suites/apis/myApiTest.js
+```
+
+## Writing Tests
+Tests are single .js files that test a group of closely related features. They are stored in `suites/apis` and `suites/portal`, with filenames matching the pattern `*Test.js`.
```

Comment:
Could you add the more high-level explanation about TestSuite (what it is, how it should look like, what it consists of). Then explain a bit more about the set of question, task, sequence, ... when you write a test suite.